openapi-model/src/main/scala/sttp/apispec/openapi/validation/OpenAPIComparator.scala

```diff
@@ -0,0 +1,284 @@
+package sttp.apispec.openapi.validation
+
+import sttp.apispec.{Schema, SchemaLike}
+import sttp.apispec.openapi.{Header, MediaType, OpenAPI, Operation, Parameter, PathItem, RequestBody, Response}
+import sttp.apispec.validation.SchemaComparator
+
+import scala.collection.immutable.ListMap
+
+class OpenAPIComparator(
+    clientOpenAPI: OpenAPI,
+    serverOpenAPI: OpenAPI
+) {
+  private val httpMethods = List("get", "post", "patch", "delete", "options", "trace", "head", "put")
+  private var serverSchemas: Map[String, Schema] = Map.empty[String, Schema]
+  private var clientSchemas: Map[String, Schema] = Map.empty[String, Schema]
+
+  def compare(): List[OpenAPICompatibilityIssue] = {
+    initSchemas()
+
+    clientOpenAPI.paths.pathItems.toList.flatMap {
+      case (pathName, clientPathItem) =>
+        val serverPathItem = serverOpenAPI.paths.pathItems.get(pathName)
+        serverPathItem match {
+          case None => Some(MissingPath(pathName))
+          case Some(serverPathItem) =>
+            val pathIssues = checkPath(pathName, clientPathItem, serverPathItem)
+            if (pathIssues.isEmpty)
+              None
+            else
+              pathIssues
+          case _ => None
+        }
+      case _ => None
+    }
+  }
+
+  private def initSchemas(): Unit = {
+    clientSchemas = clientOpenAPI.components match {
+      case Some(components) =>
+        components.schemas.flatMap {
+          case (key, schema: Schema) => Some(key, schema)
+          case _                     => None
+        }
+      case _ => Map.empty[String, Schema]
+    }
+
+    serverSchemas = serverOpenAPI.components match {
+      case Some(components) =>
+        components.schemas.flatMap {
+          case (key, schema: Schema) => Some(key, schema)
+          case _                     => None
+        }
+      case _ => Map.empty[String, Schema]
+    }
+  }
+
+  private def checkPath(
+      pathName: String,
+      clientPathItem: PathItem,
+      serverPathItem: PathItem
+  ): Option[IncompatiblePath] = {
+    val issues = httpMethods.flatMap { httpMethod =>
```

Comment:
you coudl replace the string-based logic here with a list of functions `List((_: PathItem).get, ...)`